extensions/vscode/src/commands/install-cli.ts

```diff
@@ -0,0 +1,48 @@
+const cp = require("child_process");
+
+import { window, ProgressOptions } from "vscode";
+
+/**
+ * Installs Dart Frog CLI in the user's system if not already installed.
+ */
+export const installCLI = async (): Promise<void> => {
+  if (!hasDartFrogCliInstalled()) {
+    const options: ProgressOptions = {
+      location: 15,
+      title: "Installing Dart Frog CLI...",
+    };
+    window.withProgress(options, installDartFrogCliVersion);
+  }
+};
+
+/**
+ * Whether the user has Dart Frog CLI installed in their system.
+ *
+ * @returns {boolean} True if the user has Dart Frog CLI installed in their
+ * system, false otherwise.
+ */
+function hasDartFrogCliInstalled(): boolean {
+  try {
+    cp.execSync(`dart_frog --version`);
```

Comment:
I feel that a more optimized way of doing so, would be to do a `which dart_frog`, and upon returning an empty string, we can assume that there is no cli installed.

I am not sure if on windows, `which` is the same command though. So if this suggestion makes it too complicated to handle cross platform, I am fine with the current approach.